velox/common/memory/HashStringAllocator.cpp

```diff
@@ -100,6 +104,43 @@ void HashStringAllocator::clear() {
   for (auto i = 0; i < kNumFreeLists; ++i) {
     new (&state_.freeLists()[i]) CompactDoubleList();
   }
+
+#ifdef NDEBUG
+  static const auto kHugePageSize = memory::AllocationTraits::kHugePageSize;
+  for (auto i = 0; i < state_.pool().numRanges(); ++i) {
+    const auto topRange = state_.pool().rangeAt(i);
+    const auto topRangeSize = topRange.size();
+    if (topRangeSize >= kHugePageSize) {
+      VELOX_CHECK_EQ(0, topRangeSize % kHugePageSize);
+    }
+
+    for (int64_t subRangeStart = 0; subRangeStart < topRangeSize;
+         subRangeStart += kHugePageSize) {
+      auto range = folly::Range<char*>(
+          topRange.data() + subRangeStart,
+          std::min<int64_t>(topRangeSize, kHugePageSize));
+      const auto size = range.size() - simd::kPadding;
+      auto* end = castToHeader(range.data() + size);
+      auto* header = castToHeader(range.data());
+      while (header != end) {
+        VELOX_CHECK_GE(reinterpret_cast<char*>(header), range.data());
+        VELOX_CHECK_LT(
+            reinterpret_cast<char*>(header), reinterpret_cast<char*>(end));
+        VELOX_CHECK_LE(
```

Comment:
```
VELOX_CHECK_LT(header, end);
VELOX_CHECK_LE(castToHeader(header->end()), end);
```

Reply:
I modified these two lines of code as suggested, and `error: static assertion failed: Formatting of non-void pointers is disallowed.` exception will appear during compilation.